datasource.js

```diff
@@ -340,7 +340,13 @@ function (angular, _, sdk, dateMath, kbn) {
     if (target.tags) {
       query.tags = angular.copy(target.tags);
       _.forOwn(query.tags, function(value, key) {
-        query.tags[key] = _.map(value, function(tag) { return self.templateSrv.replace(tag); });
+        query.tags[key] = _.flatten(_.map(value, function(tag) {
+          var replacement = self.templateSrv.replace(tag);
+          if (replacement.indexOf('{') == 0 && replacement.lastIndexOf('}') == replacement.length - 1) {
```

Comment:
You're right.  Is the best way to get to the data through the templateSrv.variables?  Is there a map by name somewhere?  I didn't see one but might have missed it.